tests/test_uh50_api.py

```diff
@@ -1,31 +1,69 @@
+from datetime import datetime
 import unittest
-from uh50_api.uh50 import read_uh50
-from uh50_api.uh50 import _search_data
+from uh50_api.uh50 import (UH50)
 import serial
 
 
 class UH50Test(unittest.TestCase):
 
-    def test_test(self):
+    def test_update_dummy(self):
+        heat_meter = UH50('test')
+        heat_meter.update_dummy()      
+        self.assertRegex(heat_meter.heat_usage_gj, r'^999.\d{3}$')
+        self.assertRegex(heat_meter.volume_usage_m3, r'^9999.\d{2}$')
+        self.assertEqual('LUGCUH50', heat_meter.model)
+
+    def test_validate_fail(self):
+        heat_meter = UH50('non-existing-port')
+        with self.assertRaises(serial.serialutil.SerialException):
+            _ = heat_meter.validate()
+
+    def test_connect_fail(self):
+        heat_meter = UH50('non-existing-port')
         with self.assertRaises(serial.serialutil.SerialException):
-            _ = read_uh50('non-existing-port')
+            _ = heat_meter._connect_serial()
 
-    def test_search_data_list(self):
-        gj, m3 = _search_data(['6.8(0255.987*GJ)6.26(02458.16*m3)9.21(66153690)', \
-                            '6.26*01(02196.39*m3)6.8*01(0233.431*GJ)', \
-                            'F(0)9.20(66153690)6.35(60*m)', \
-                            '6.6(0022.4*kW)6.6*01(0022.4*kW)6.33(000.708*m3ph)9.4(098.5*C&096.1*C)',])
-        self.assertEqual('0255.987', gj)
-        self.assertEqual('02458.16', m3)
+    def test_update_fail(self):
+        heat_meter = UH50('non-existing-port')
+        with self.assertRaises(serial.serialutil.SerialException):
+            _ = heat_meter.update()
 
-    def test_search_data_str(self):
-        gj, m3 = _search_data('6.8(0255.987*GJ)6.26(02458.16*m3)9.21(66153690)')
-        self.assertEqual('0255.987', gj)
-        self.assertEqual('02458.16', m3)
+    def test_search_response(self):
```

Comment:
Testcase faalt op wat een compile error had moeten zijn.

Reply:
waar doel je op? Bij mij slagen alle tests ...

Reply:
Sorry, ik gooide twee fouten op 1 hoop waar ik dat niet had moeten doen. Deze faalt bij mij op het inlezen van de dummy_response.txt

Error
Traceback (most recent call last):
  File "C:\Projects\uh50\tests\test_uh50_api.py", line 32, in test_search_response
    f = open('tests/dummy_response.txt')
FileNotFoundError: [Errno 2] No such file or directory: 'tests/dummy_response.txt

Reply:
van waaruit run jij die test dan? Dit gaat goed: 
(venv) PS C:\repositories\thuis\domotica\uh50> python3 -m unittest tests/test_uh50_api.py

Reply:
Ahja, ik runde m met pycharm direct uit die folder. Dan klopt het pad natuurlijk niet. Het is op zn minst een "flaky test". 
f = open(Path(__file__).with_name("dummy_response.txt")) Dit is een manier om een fil te openen relatief tov de huidige python file (middels __file__).